frontend/packages/sdk/src/pruntime/createPruntimeClient.ts

```diff
@@ -0,0 +1,37 @@
+import { fetch } from 'undici'
+import { prpc, pruntime_rpc as pruntimeRpc } from './proto'
+
+/**
+ * Create a http client prepared for protobuf
+ *
+ * @param baseURL The base URL of the pruntime server
+ * @returns A PhactoryAPI object
+ */
+export default function createPruntimeClient(baseURL: string) {
+  const pruntimeApi = pruntimeRpc.PhactoryAPI.create(async (method, requestData, callback) => {
+    try {
+      const resp = await fetch(`${baseURL}/prpc/PhactoryAPI.${method.name}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/octet-stream',
+        },
+        body: new Uint8Array(requestData),
+      })
+      if (resp.status === 200) {
+        const buffer = await (await resp.blob()).arrayBuffer()
+        callback(null, new Uint8Array(buffer))
+      } else if (resp.status === 500) {
+        const buffer = await (await resp.blob()).arrayBuffer()
+        const prpcError = prpc.PrpcError.decode(new Uint8Array(buffer))
+        throw new Error(`PrpcError: ${prpcError.message}`)
+      } else {
+        throw new Error(`Unexpected Response Status: ${resp.status}`)
```

Comment:
Attach the resp to the error for reference?

Reply:
The text would be very large when it returns as an HTML page. For example, Cloudflare returns a page for 522.

Reply:
Yes. No need to add the body in the message, but attach to an error field like this:

```js
new Error("Error message", { details: "http error" });
```